offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/OfferTemplate.groovy

```diff
@@ -0,0 +1,370 @@
+        private final String unit
+
+        static OfferPosition createProductItem(int position, OfferItem offerItem) {
+            String name = offerItem.getProductName()
+            String description = createProductItemDescription(offerItem.getProductDescription(), offerItem.getServiceProvider())
+            double quantity = offerItem.getQuantity()
+            double unitPrice = offerItem.getUnitPrice()
+            double total = offerItem.getItemTotal()
+            String unit = offerItem.getUnit()
+            return new OfferPosition(position, name, description, quantity, unitPrice, total, unit)
+        }
+
+        static OfferPosition createDiscount(int position, int discountedPosition, OfferItem offerItem) {
+            String name = "Discount"
+            double quantity = offerItem.getQuantity()
+            double unitPrice = offerItem.getDiscountPerUnit() * (-1)
+            double total = offerItem.getQuantityDiscount() * (-1)
+            String unit = offerItem.getUnit()
+            double discountPercentage = offerItem.getDiscountPercentage()
+            String description = createDiscountDescription(discountedPosition, quantity, unit, discountPercentage)
+            return new OfferPosition(position, name, description, quantity, unitPrice, total, unit)
+        }
+
+        private OfferPosition(int position, String name, String description, double quantity, double unitPrice, double total, String unit) {
+            this.position = position
+            this.name = name
+            this.description = description
+            this.quantity = quantity
+            this.unitPrice = unitPrice
+            this.total = total
+            this.unit = unit
+        }
+
+        /**
+         * Generates a html representation of this offer position
+         * @return a html string representing this position
+         * @since 1.1.0
+         */
+        String outerHtml() {
+            String quantity = this.quantity.toString()
+            String unitPrice = Currency.format(this.unitPrice)
+            String total = Currency.format(this.total)
+            String html = """\
+            <tr>
+              <td class="item-number-column">${position}</td>
+              <td class="item-description-column">
+                <div class="item-title">${name}</div>
+              </td>
+              <td class="item-amount-column">${quantity}</td>
+              <td class="item-unit-column">${unit}</td>
+              <td class="item-unit-price-column">${unitPrice}</td>
+              <td class="item-total-column">${total}</td>
+            </tr>
+            <tr>
+              <td class="item-number-column"></td>
+              <td class="item-description-column" colspan="3">
+                <div class="item-description">
+                  ${description}
+                </div>
+              </td>
+            </tr>\
+            """.stripIndent()
+            return html
+        }
```

Comment:
I like that! Looks very clean and easy to read!

Reply:
This clarity can only be achieved because we have the fields holding relevant information.
This is why I introduced the OfferPosition